### PR TITLE
make thumb size in folder contents listing adjustable/supressable

### PR DESCRIPTION
**Pull request by fgrcon**
(uses site control panel settings: thumb_size_tables ...)
  https://github.com/plone/Products.CMFPlone/issues/1734

### CODE REVIEW

File: plone/app/content/browser/vocabulary.py [hunk 1]
```diff
@@ -233,8 +237,22 @@ def __call__(self):
                             _(safe_unicode(val)),
                             context=self.request
                         )
+
                     else:
                         item[key] = val
+                    if key == 'getMimeIcon':
+                        item[key] = None
+                        if  vocab_item.portal_type =='File':
```
Comment (jensens) [inline plone/app/content/browser/vocabulary.py]: I don't like it is bound to type File only.

File: plone/app/content/browser/vocabulary.py [hunk 2]
```diff
@@ -233,8 +237,22 @@ def __call__(self):
...
+                            #get mime type icon url from mimetype registry'
+                            portal_url = api.portal.get().absolute_url()
```
Comment (jensens) [inline plone/app/content/browser/vocabulary.py]: use ``plone.api.portal.get_navigation_root`` instead

File: plone/app/content/browser/vocabulary.py [hunk 3]
```diff
@@ -233,8 +237,22 @@ def __call__(self):
...
+                            mtt = api.portal.get_tool(
+                                        name='mimetypes_registry')
+                            if vocab_item.getObject().file.contentType:
```
Comment (jensens) [inline plone/app/content/browser/vocabulary.py]: we really need the mimetype as catalog metadata.

Comment (fgrcon) [inline plone/app/content/browser/vocabulary.py]: thats why i proposed https://github.com/plone/Products.CMFPlone/issues/1995. I suggest to merge 1995 first. (upgrade changes are interdependent)

File: plone/app/content/browser/vocabulary.py (hunk 2 above)
Comment (gforcada) [inline plone/app/content/browser/vocabulary.py]: No plone.api should be used actually...

File: plone/app/content/browser/contents/__init__.py [hunk 4]
```diff
@@ -168,6 +170,7 @@ def ignored_columns(self):
             'effective',
             'expires',
             'getIcon',
+			'getMimeIcon',
```
Comment (gforcada) [inline plone/app/content/browser/contents/__init__.py]: would you mind aligning  this?

File: plone/app/content/browser/vocabulary.py [hunk 5]
```diff
@@ -233,8 +238,33 @@ def __call__(self):
...
+                            ####v with:
+                            #contenttype = aq_base(getattr(self, 'mime_type', None))
+                            ####^ with
+                            mtt = api.portal.get_tool(
+                                        name='mimetypes_registry')
+                            ctype = mtt.lookup(contenttype)
+                            item[key] = os.path.join(
+                                    portal_url,
+                                    guess_icon_path(ctype[0]))
+                        except:
```
Comment (gforcada) [inline plone/app/content/browser/vocabulary.py]: no bare exceptions please

File: plone/app/content/browser/vocabulary.py (hunk 2 above)
Comment (fgrcon) [inline plone/app/content/browser/vocabulary.py]: ok removed api and use naroot

File: plone/app/content/browser/contents/__init__.py (hunk 4 above)
Comment (fgrcon) [inline plone/app/content/browser/contents/__init__.py]: fixed( (tabs ...)

**Pull request merged**